core/state/statedb_test.go

```diff
@@ -727,3 +741,142 @@ func TestMissingTrieNodes(t *testing.T) {
 		t.Fatalf("expected error, got root :%x", root)
 	}
 }
+
+func TestStateDBAccessList(t *testing.T) {
+	// Some helpers
+	addr := func(a string) common.Address {
+		return common.HexToAddress(a)
+	}
+	slot := func(a string) common.Hash {
+		return common.HexToHash(a)
+	}
+
+	memDb := rawdb.NewMemoryDatabase()
+	db := NewDatabase(memDb)
+	state, _ := New(common.Hash{}, db, nil)
+	state.accessList = NewAccessList()
+
+	verifyAddrs := func(a ...string) {
+		t.Helper()
+		for _, address := range a {
+			if !state.AddrInAccessList(addr(address)) {
+				t.Fatalf("expected %x to be in access list", a)
+			}
+		}
```

Comment:
I'd expand this to check the reverse, that no accounts from the access list are missing from a. Otherwise you're not checking for extra items.